internal/build/lifecycle_execution.go

```diff
@@ -228,10 +234,25 @@ func (l *LifecycleExecution) Run(ctx context.Context, phaseFactoryCreator PhaseF
 
 		group, _ := errgroup.WithContext(context.TODO())
 		if l.platformAPI.AtLeast("0.10") && l.hasExtensionsForBuild() {
-			group.Go(func() error {
-				l.logger.Info(style.Step("EXTENDING (BUILD)"))
-				return l.ExtendBuild(ctx, buildCache, phaseFactory)
-			})
+			if l.opts.Publish {
+				group.Go(func() error {
+					l.logger.Info(style.Step("EXTENDING (BUILD)"))
+					return l.ExtendBuild(ctx, buildCache, phaseFactory)
+				})
+			} else {
```

Comment:
```suggestion
```
We should be able to remove this block. We can extend the build image using the daemon even when we are publishing the app image to a registry

Reply:
done